src/main.cpp

```diff
@@ -282,105 +277,88 @@ static std::optional<struct timespec> get_delta_taitime()
-      }))
-    return false;
-
-  if (!get_bool_env_var("BPFTRACE_DEBUG_OUTPUT",
-                        [&](bool x) { bpftrace.debug_output_ = x; }))
-    return false;
-
-  if (!get_bool_env_var("BPFTRACE_LAZY_SYMBOLICATION", [&](bool x) {
-        config_setter.set(ConfigKeyBool::lazy_symbolication, x);
-      }))
-    return false;
-
-  if (!get_uint64_env_var("BPFTRACE_MAX_MAP_KEYS", [&](uint64_t x) {
-        config_setter.set(ConfigKeyInt::max_map_keys, x);
-      }))
-    return false;
-
-  if (!get_uint64_env_var("BPFTRACE_MAX_PROBES", [&](uint64_t x) {
-        config_setter.set(ConfigKeyInt::max_probes, x);
-      }))
-    return false;
-
-  if (!get_uint64_env_var("BPFTRACE_MAX_BPF_PROGS", [&](uint64_t x) {
-        config_setter.set(ConfigKeyInt::max_bpf_progs, x);
-      }))
-    return false;
-
-  if (!get_uint64_env_var("BPFTRACE_LOG_SIZE", [&](uint64_t x) {
-        config_setter.set(ConfigKeyInt::log_size, x);
-      }))
-    return false;
-
-  if (!get_uint64_env_var("BPFTRACE_PERF_RB_PAGES", [&](uint64_t x) {
-        config_setter.set(ConfigKeyInt::perf_rb_pages, x);
-      }))
-    return false;
-
-  if (!get_uint64_env_var("BPFTRACE_MAX_TYPE_RES_ITERATIONS", [&](uint64_t x) {
-        config_setter.set(ConfigKeyInt::max_type_res_iterations, x);
-      }))
-    return false;
-
-  if (!get_uint64_env_var("BPFTRACE_MAX_CAT_BYTES", [&](uint64_t x) {
-        config_setter.set(ConfigKeyInt::max_cat_bytes, x);
-      }))
-    return false;
+  get_bool_env_var("BPFTRACE_CPP_DEMANGLE", [&](bool x) {
+    config_setter.set(ConfigKeyBool::cpp_demangle, x);
+  });
+
+  get_bool_env_var("BPFTRACE_DEBUG_OUTPUT",
+                   [&](bool x) { bpftrace.debug_output_ = x; });
+
+  get_bool_env_var("BPFTRACE_LAZY_SYMBOLICATION", [&](bool x) {
+    config_setter.set(ConfigKeyBool::lazy_symbolication, x);
+  });
+
+  get_uint64_env_var("BPFTRACE_MAX_MAP_KEYS", [&](uint64_t x) {
+    config_setter.set(ConfigKeyInt::max_map_keys, x);
+  });
+
+  get_uint64_env_var("BPFTRACE_MAX_PROBES", [&](uint64_t x) {
+    config_setter.set(ConfigKeyInt::max_probes, x);
+  });
+
+  get_uint64_env_var("BPFTRACE_MAX_BPF_PROGS", [&](uint64_t x) {
+    config_setter.set(ConfigKeyInt::max_bpf_progs, x);
+  });
+
+  get_uint64_env_var("BPFTRACE_LOG_SIZE", [&](uint64_t x) {
+    config_setter.set(ConfigKeyInt::log_size, x);
+  });
+
+  get_uint64_env_var("BPFTRACE_PERF_RB_PAGES", [&](uint64_t x) {
+    config_setter.set(ConfigKeyInt::perf_rb_pages, x);
+  });
+
+  get_uint64_env_var("BPFTRACE_MAX_TYPE_RES_ITERATIONS", [&](uint64_t x) {
+    config_setter.set(ConfigKeyInt::max_type_res_iterations, x);
+  });
+
+  get_uint64_env_var("BPFTRACE_MAX_CAT_BYTES", [&](uint64_t x) {
+    config_setter.set(ConfigKeyInt::max_cat_bytes, x);
+  });
 
   if (const char* env_p = std::getenv("BPFTRACE_CACHE_USER_SYMBOLS")) {
     const std::string s(env_p);
     if (!config_setter.set_user_symbol_cache_type(s))
-      return false;
+      exit(1);
```

Comment:
Why not `LOG(FATAL)` in `ConfigSetter::set_user_symbol_cache_type` and get rid of the `if` and `exit(1)`? Same below for `set_stack_mode`.

Reply:
Because then I can't easily test this code path in unit tests (AFAIK).